src/commands/deploy.js

```diff
@@ -55,26 +55,31 @@ deploy.deploy = function(api, params) {
 };
 
 deploy.restart = function(api, params) {
-  var alias = params.options.alias;
-  var quiet = params.options.quiet;
-  var commitId = params.options.commit;
-  var withoutCache = params.options["without-cache"];
-
-  var s_appData = AppConfig.getAppData(alias).toProperty();
-  var s_commitId = s_appData.flatMapLatest(function(app_data) {
-    return Application.get(api, app_data.app_id, app_data.app_orga);
-  }).flatMapLatest(function(app) {
-    return app.commitId;
-  });
-
-  var s_deploy = s_appData.flatMapLatest(function(app_data) {
-    var suffix = "";
-    if(commitId) suffix += " on commit #" + commitId;
-    if(withoutCache) suffix += " without using cache";
-
-    Logger.println("Restarting " + app_data.name + suffix);
-    return Application.redeploy(api, app_data.app_id, app_data.org_id, commitId, withoutCache);
-  }).toProperty();
+  const { alias, quiet, commit, 'without-cache': withoutCache } = params.options
+
+  const s_fullCommitId = Git.resolveFullCommitId(commit)
+  s_fullCommitId.onError((e) => {
+      Logger.error(`Commit id ${commit} is ambiguous`)
+      process.exit(1);
+    })
```

Comment:
I try to avoid early returns, would it be possible to return an error stream and let the error be managed only at the end?